pkg/controller/common/reconciler/results.go

```diff
@@ -97,7 +99,7 @@ func (r *Results) HasRequeue() bool {
 	if r == nil {
 		return false
 	}
-	return r.currResult.Result.Requeue || r.currResult.Result.RequeueAfter > 0
+	return r.currResult.Result.Requeue || r.currResult.Result.RequeueAfter > 0 //nolint:staticcheck
```

Comment:
Are we preserving `Requeue` here because this project can be used as a library, or because `Requeue` is still used in some places in the project's code? (or both)

Reply:
Yes kind of. I was thinking of this as some sort of library function where we have to support `Requeue` as long as the field exists in the struct.